src/main/java/seedu/academydirectory/model/student/Assessment.java

```diff
@@ -0,0 +1,68 @@
+package seedu.academydirectory.model.student;
+
+import java.util.HashMap;
+
+/**
+ * Represents a Student's assessment in the address book.
+ */
+public class Assessment implements Information {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Grades can only be recorded for the following assessments: RA1, MIDTERM, RA2, PE, FINAL.";
+
```

Comment:
For extensibility purposes, do you think it's better to not hard code the possible types of assessments here? For CS1101S these are all I guess but I'm thinking if it might be better to abstract these out as subclasses of assessments. A counter-argument would be that that's over-engineering things..

Reply:
Initially my thought was to allow tutor to add the assessment that they want to record individually. But like what you mentioned, these assessments are fixed and I believe fixing the assessments will allow us to work on the analysis of the grades more easily in the future.

But yes, I see the benefits in abstracting them out, because we can perform the analysis within the various assessment classes too. Will look into this further!